interpreter/operator_arithmetic.go

```diff
@@ -945,7 +1027,7 @@ func maxValue(t types.IType, evaluationTimestamp *time.Time) (result.Value, erro
 		if evaluationTimestamp == nil {
 			return result.Value{}, fmt.Errorf("internal error - evaluation timestamp cannot be nil for DateTime max value")
 		}
-		return result.New(result.DateTime{Date: time.Date(9999, 12, 31, 23, 59, 59, 999, evaluationTimestamp.Location()), Precision: model.MILLISECOND})
+		return result.New(result.DateTime{Date: time.Date(9999, 12, 31, 23, 59, 59, 999000000, time.UTC), Precision: model.MILLISECOND})
```

Comment:
This should be reverted (and the associated test as well).
See the comments at the top of the function, the tldr is that this is a case that engines have leeway on, we chose to go this route to make handling some edge cases for datetime evaluation more intuitive / easy to do.